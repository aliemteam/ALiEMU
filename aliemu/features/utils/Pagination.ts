// tslint:disable export-name
/**
 * Takes an array and returns an array containing the object params that should
 *   be visible on the current page.
 *
 * @param rows Any array of data to be paginated.
 * @param visibleRows Number of rows that should be visible.
 * @param currentPage Current selected page.
 * @return A filtered, paginated array.
 */
export function paginate(rows: any[], visibleRows: number, currentPage: number): any[] {
    return rows.filter((_uid, i: number) =>
        (visibleRows * currentPage) <= i && i < (visibleRows * currentPage + visibleRows)
    );
}
