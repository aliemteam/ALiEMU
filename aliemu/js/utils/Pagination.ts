// tslint:disable export-name
/**
 * Takes an array and returns an array containing the object params that should
 *   be visible on the current page.
 *
 * @param rows Any array of data to be paginated.
 * @param visibleRows Number of rows that should be visible.
 * @param currentPage Current selected page.
 * @return A filtered, paginated array.
 */
export function paginate<T>(rows: T[], visibleRows: number, currentPage: number): T[] {
    return rows.filter((_uid, i: number) => (
        (visibleRows * currentPage) <= i && i < (visibleRows * currentPage + visibleRows)
    ));
}
