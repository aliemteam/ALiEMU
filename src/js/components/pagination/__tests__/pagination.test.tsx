import React from 'react';
import renderer from 'react-test-renderer';

import Pagination from '../index';

describe('<Pagination />', () => {
    beforeAll(() => {
        Date.now = jest.fn(() => 0);
    });
    it('should render correctly', () => {
        const tree = renderer.create(
            <Pagination value={3} total={5} onChange={jest.fn()} />,
        );
        expect(tree).toMatchSnapshot();
    });
});
