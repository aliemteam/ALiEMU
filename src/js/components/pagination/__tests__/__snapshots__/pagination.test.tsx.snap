// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render correctly 1`] = `
<nav
  aria-labelledby="pagination-0"
  className="nav alignRight"
  role="navigation"
>
  <div
    className="pager"
  >
    <span
      aria-live="polite"
      role="status"
    >
      <h2
        className="clipped"
        id="pagination-0"
      >
        Table Pagination - Page 3
      </h2>
    </span>
    <button
      aria-label="previous page"
      className="buttonPrev"
      data-page={2}
      disabled={false}
      onClick={[MockFunction]}
      style={
        Object {
          "fontSize": 15,
          "height": 30,
          "minWidth": 30,
        }
      }
    >
      <i
        className="material-icons"
        style={
          Object {
            "color": undefined,
            "fontSize": 18,
          }
        }
      >
        chevron_left
      </i>
    </button>
    <div
      className="pageList"
      role="list"
    >
      <button
        aria-label="page 1"
        aria-posinset={1}
        aria-setsize={5}
        className="button"
        data-page={1}
        onClick={[MockFunction]}
        role="listitem"
        style={
          Object {
            "fontSize": 15,
            "height": 30,
            "minWidth": 30,
          }
        }
      >
        1
      </button>
      <button
        aria-label="page 2"
        aria-posinset={2}
        aria-setsize={5}
        className="button"
        data-page={2}
        onClick={[MockFunction]}
        role="listitem"
        style={
          Object {
            "fontSize": 15,
            "height": 30,
            "minWidth": 30,
          }
        }
      >
        2
      </button>
      <button
        aria-current="page"
        aria-label="page 3"
        aria-posinset={3}
        aria-setsize={5}
        className="button"
        data-page={3}
        onClick={[MockFunction]}
        role="listitem"
        style={
          Object {
            "fontSize": 15,
            "height": 30,
            "minWidth": 30,
          }
        }
      >
        3
      </button>
      <button
        aria-label="page 4"
        aria-posinset={4}
        aria-setsize={5}
        className="button"
        data-page={4}
        onClick={[MockFunction]}
        role="listitem"
        style={
          Object {
            "fontSize": 15,
            "height": 30,
            "minWidth": 30,
          }
        }
      >
        4
      </button>
      <button
        aria-label="page 5"
        aria-posinset={5}
        aria-setsize={5}
        className="button"
        data-page={5}
        onClick={[MockFunction]}
        role="listitem"
        style={
          Object {
            "fontSize": 15,
            "height": 30,
            "minWidth": 30,
          }
        }
      >
        5
      </button>
    </div>
    <button
      aria-label="next page"
      className="buttonNext"
      data-page={4}
      disabled={false}
      onClick={[MockFunction]}
      style={
        Object {
          "fontSize": 15,
          "height": 30,
          "minWidth": 30,
        }
      }
    >
      <i
        className="material-icons"
        style={
          Object {
            "color": undefined,
            "fontSize": 18,
          }
        }
      >
        chevron_right
      </i>
    </button>
  </div>
</nav>
`;
